### PR TITLE
flamebench: streamline profiling benches 🔥🪑™️

### PR DESCRIPTION
**Pull request by AaronO**
Introducing `flamebench` a small internal utility to facilitate profiling, providing devs with a one-liner command to get a flamegraph from benches. (See https://github.com/denoland/deno/pull/10223 for `bench_util` stuff)

## Usage

```
❯ ./tools/flamebench.js ls
op_baseline
ser
de
```

```
❯ ./tools/flamebench.js op_baseline async

...

Profiling bench_op_async

...

writing flamegraph to "~/git/deno/flamebench.svg"
```

### CODE REVIEW

File: tools/flamebench [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
```
Comment (bartlomieju) [inline tools/flamebench:1]: Could this script be written in Deno instead of bash?

Comment (AaronO) [reply tools/flamebench]: Done

**Comment by AaronO** [comment]
FYI, I rewrote this from `bash` to `js` and it's ~2x longer, but probably easier to grok and maintain. There's still a good amount of shelling out to bash, fs walking and co should be doable in JS.

File: tools/flamebench.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  } catch (_) {
+    return false;
+  }
+}
+
+async function main() {
+  const { 0: benchName, 1: benchFilter } = Deno.args;
+  // Print usage if no bench specified
+  if (!benchName) {
+    console.log("flamebench <bench_name> [bench_filter]");
```
Comment (bartlomieju) [inline tools/flamebench.js:76]: Could you display of available benches here as well?

**Review by bartlomieju — COMMENTED** [review]
I've tried to run locally:
```
❯ ./tools/flamebench.js op_baseline
    Finished release [optimized] target(s) in 0.33s
dtrace: description 'profile-997 ' matched 1 probe

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured

dtrace: pid 29231 has exited
dtrace: error on enabled probe ID 1 (ID 43: profile:::profile-997): invalid address (0x7ffee0aa1730) in action #2
writing flamegraph to "/dev/deno/flamebench.svg"
thread 'main' panicked at 'unable to generate a flamegraph from the collapsed stack data: Io(Custom { kind: InvalidData, error: "No stack counts found" })', /.cargo/registry/src/github.com-1ecc6299db9ec823/flamegraph-0.4.0/src/lib.rs:315:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Could add an entry to `tools/README.md` with short instruction how to use this script?

File: tools/flamebench.js (hunk 2 above)
Comment (AaronO) [reply tools/flamebench.js:76]: Done:
```
❯ ./tools/flamebench.js
flamebench <bench_name> [bench_filter]

Available benches:
op_baseline
ser
de
```

**Comment by AaronO** [comment]
> I've tried to run locally:
> 
> ```
> ❯ ./tools/flamebench.js op_baseline
>     Finished release [optimized] target(s) in 0.33s
> dtrace: description 'profile-997 ' matched 1 probe
> 
> running 0 tests
> 
> test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured
> 
> dtrace: pid 29231 has exited
> dtrace: error on enabled probe ID 1 (ID 43: profile:::profile-997): invalid address (0x7ffee0aa1730) in action #2
> writing flamegraph to "/dev/deno/flamebench.svg"
> thread 'main' panicked at 'unable to generate a flamegraph from the collapsed stack data: Io(Custom { kind: InvalidData, error: "No stack counts found" })', /.cargo/registry/src/github.com-1ecc6299db9ec823/flamegraph-0.4.0/src/lib.rs:315:6
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```
> 
> Could add an entry to `tools/README.md` with short instruction how to use this script?

This should be fixed, it was stringifying the `benchFilter` to `"undefined"` which didn't match any benches, so the process was very short lived and causing `flamegraph` to crash since there was nothing meaningful for it to sample.

I usually run it with a filter / sub-bench, which is why I hadn't noticed this regression after porting it over from bash in my own usage.

**Comment by AaronO** [comment]
@bartlomieju Added an entry in `tools/README.md`, if that's good this should be good to land IMO

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**